### PR TITLE
vkd3d-proton: update and adopt

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

Splitted from https://github.com/NixOS/nixpkgs/pull/294280

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AndersonTorres** [comment]
@SuperSandro2000 it is really needed to devendor this?
I did not end it completely yet, but there are three submodules and one of them has another three extra submodules.

**Comment by AndersonTorres** [comment]
@SuperSamus here is vkd3d-proton, with updates!

File: pkgs/by-name/vk/vkd3d-proton/package.nix [hunk 1]
```diff
@@ -32,10 +32,12 @@ stdenv.mkDerivation (finalAttrs: {
...
-    # these are both embedded in the output files
-    # git describe --tags
-    shortRev = builtins.substring 0 8 finalAttrs.src.rev;
-    realVersion = "v2.8-302-g${shortRev}";
+    # these are both embedded in the output files; they are obtained via
+    # git describe --always --exclude='*' --abbrev=15 --dirty=0
+    # git describe --always --tags --dirty=+
+    rev = "105b5b77c9a34fd336b5c604e3c7a6cc48f39c3a";
+    shortRev = lib.substring 0 8 rev;
+    realVersion = finalAttrs.src.rev;
```
Comment (SuperSandro2000) [inline pkgs/by-name/vk/vkd3d-proton/package.nix]:
```suggestion
```
we might as well use this directly below

File: pkgs/by-name/vk/vkd3d-proton/package.nix [hunk 2]
```diff
@@ -31,21 +25,18 @@ stdenv.mkDerivation (finalAttrs: {
...
 
-  prePatch = let
-    # these are both embedded in the output files; they are obtained via
-    # vkd3dBuild = $(git describe --always --exclude='*' --abbrev=15 --dirty=0)
-    # vkd3dVersion = $(git describe --always --tags --dirty=+)
-    inherit (finalAttrs.src) rev; # Change this to raw rev when needed
-    vkd3dBuild = lib.substring 0 15 rev;
-    vkd3dVersion = "v2.11.1-135-g${lib.substring 0 8 rev}";
-  in ''
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/by-name/vk/vkd3d-proton/package.nix:28]: This should be postPatch

File: pkgs/by-name/vk/vkd3d-proton/sources.nix [hunk 3]
```diff
@@ -5,11 +5,11 @@
 {
   vkd3d-proton = {
     pname = "vkd3d-proton";
-    version = "2.11.1-unstable-2024-03-07";
+    version = "2.12";
     src = fetchFromGitHub {
       owner = "HansKristian-Work";
       repo = "vkd3d-proton";
-      rev = "ffb04fb7091f932ca95da236d63ab9a4064d6443";
+      rev = "v2.12";
```
Comment (SuperSandro2000) [inline pkgs/by-name/vk/vkd3d-proton/sources.nix:12]:
```suggestion
      rev = "v${version}";
```

**Review by SuperSandro2000 — COMMENTED** [review]
Can you cleanup the history a bit and combine the refactors? When reviewing commit by commit, I got confused and only saw latter that some comments where already fixed.

File: pkgs/by-name/vk/vkd3d-proton/sources.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/by-name/vk/vkd3d-proton/sources.nix:13]: I am still figuring out how to male this in a not-so-ugly way.

File: pkgs/by-name/vk/vkd3d-proton/package.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    wine
+  ];
+
+  strictDeps = true;
+
+  postPatch = ''
+    substituteInPlace meson.build \
+      --replace "vkd3d_build = vcs_tag(" \
+                "vkd3d_build = vcs_tag( fallback : '$(cat .nixpkgs-auxfiles/vkd3d_build)'", \
+      --replace "vkd3d_version = vcs_tag(" \
```
Comment (SuperSandro2000) [inline pkgs/by-name/vk/vkd3d-proton/package.nix:32]:
```suggestion
      --replace-fail "vkd3d_build = vcs_tag(" \
                "vkd3d_build = vcs_tag( fallback : '$(cat .nixpkgs-auxfiles/vkd3d_build)'", \
      --replace-fail "vkd3d_version = vcs_tag(" \
```

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

File: pkgs/by-name/vk/vkd3d-proton/package.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/by-name/vk/vkd3d-proton/package.nix]: Done.

**Comment by AndersonTorres** [comment]
BYE

**Pull request closed without merging**